src/hooks/useGlobalAppState.ts

```diff
@@ -1,116 +1,51 @@
-  devlog('tryLockingApp with', { parsedRoute, client })
-
-  return await determineSecurityFlow(
-    client,
-    // Let's assume the parsedRoute could be undefined for unknown reasons
-    // In that case we just don't pass the navigation object and the security flow will default to home
-    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-    parsedRoute?.name !== 'default'
-      ? {
-          navigation: RootNavigation.navigationRef as NavigationContainerRef<
-            Record<string, unknown>
-          >,
-          href: parsedRoute.params.href,
-          slug: parsedRoute.params.slug
-        }
-      : undefined
-  )
-}
+initSharingMode()
 
 const handleSleep = (): void => {
   showSplashScreen()
-    .then(() => storeData(StorageKeys.LastActivity, Date.now().toString()))
+    .then(async () => {
+      setIsSecurityFlowPassed(false)
+      setFilesToUpload([])
+      return await storeData(StorageKeys.LastActivity, Date.now().toString())
+    })
     .catch(reason => log.error('Failed when going to sleep', reason))
 }
 
-/**
- * If we went to sleep while on the lock screen, we don't want to lock the app
- *
- * If we went to sleep on an iOS device, we don't want to check the timer and always autolock the app
- *
- * In any other case, we just check the inactivity timer and ask to lock the app if needed
- */
-export const _shouldLockApp = (
-  parsedRoute: Route<string>,
-  timeSinceLastActivity: number
-): boolean => {
-  try {
-    // Accessing a property of parsedRoute will throw an error if it's null or undefined
-    if (parsedRoute.name === routes.lock) return false
-  } catch (error) {
-    // If an error is thrown (i.e., parsedRoute is null or undefined), we default to locking the app
-    return true
-  }
-
-  if (timeSinceLastActivity < 0) return true
-
-  if (Platform.OS === 'ios') return true
-
-  return timeSinceLastActivity > TIMEOUT_VALUE
-}
-
 const handleWakeUp = (client: CozyClient): void => {
-  const currentRoute = RootNavigation.navigationRef.getCurrentRoute()
-  const parsedRoute = JSON.parse(JSON.stringify(currentRoute)) as Route<
-    string,
-    { href: string; slug: string }
-  >
-
-  getData<string>(StorageKeys.LastActivity)
-    .then((lastActivity): number => {
-      const now = Date.now()
-      const lastActivityDate = new Date(parseInt(lastActivity ?? '0', 10))
-
-      return now - lastActivityDate.getTime()
-    })
-    .then(timeSinceLastActivity => {
-      if (_shouldLockApp(parsedRoute, timeSinceLastActivity))
-        return tryLockingApp(parsedRoute, client)
-      else {
-        devlog(
-          'handleWakeUp: no need check the security status, hiding splash screen'
-        )
-        return hideSplashScreen()
-      }
-    })
-    .catch(reason => log.error('Failed when waking up', reason))
+  handleSecurityFlowWakeUp(client)
+
+  if (isSharingMode()) {
+    log.info('useGlobalAppState: handleWakeUp, sharing mode')
+    navigate(routes.sharing)
```

Comment:
As `handleSecurityFlowWakeUp` has async side effect that call `navigate`. Does this means that `navigate(routes.sharing)` will have race condition with `handleSecurityFlowWakeUp`?

Reply:
I didn't have issues because it redirects to home then sharing in sequence. I shall look into possible errors

Reply:
any race condition should now be handled since we await for securityFlow end on wakeup